content/300-guides/100-performance-and-optimization/090-metrics.mdx

```diff
@@ -0,0 +1,416 @@
+  - `query_operation_total_elapsed_time_ms`: The execution time for all operations.
+
+You can [add global labels to your metrics data](#global-labels) to help you group and separate your metrics, for example by infrastructure region or server.
+
+## Prerequisites
+
+To use Prisma metrics, you must do the following:
+
+1. [Install the appropriate dependencies](#step-1-install-up-to-date-prisma-dependencies).
+1. [Enable the `metrics` feature flag in your Prisma schema file](#step-2-enable-the-feature-flag-in-the-prisma-schema-file).
+
+### Step 1. Install up-to-date Prisma dependencies
+
+Use version `3.15.0` or higher of the `prisma` and `@prisma/client` npm packages.
+
+```terminal
+npm install prisma@latest --save-dev
+npm install @prisma/client@latest --save
+```
+
+### Step 2: Enable the feature flag in the Prisma schema file
+
+In the `generator` block of your `schema.prisma` file, enable the `metrics` feature flag:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["metrics"]
+}
+```
+
+## Retrieve metrics in JSON format
+
+When you retrieve metrics in JSON format, you can use them in the format they are returned, or [send them to StatSD](#use-prisma-metrics-with-statsd) to visualize how they change over time.
+
+To retrieve metrics in JSON format, add the following lines to your application code:
+
+```ts
+let metrics = await prisma.$metrics.json()
+console.log(metrics)
+```
+
+This returns metrics as follows:
+
+```json
+{
+  "counters": [
+    {
+      "key": "query_total_operations",
+      "labels": {},
+      "value": 0,
+      "description": "The total number of operations executed"
+    },
+    {
+      "key": "query_total_queries",
+      "labels": {},
+      "value": 0,
+      "description": "The total number of queries executed"
+    }
+  ],
+  "gauges": [
+    {
+      "key": "pool_active_connections",
+      "labels": {},
+      "value": 0,
+      "description": "The number of active connections in use"
+    },
+    {
+      "key": "pool_idle_connections",
+      "labels": {},
+      "value": 21,
+      "description": "The number of connections that are not being used"
+    },
+    {
+      "key": "pool_wait_count",
+      "labels": {},
+      "value": 0,
+      "description": "The number of workers waiting for a connection"
+    },
+    {
+      "key": "query_active_transactions",
+      "labels": {},
+      "value": 0,
+      "description": "The number of active transactions"
+    }
+  ],
+  "histograms": [
+    {
+      "key": "pool_wait_duration_ms",
+      "labels": {},
+      "value": {
+        "buckets": [
+          [0, 0],
+          [1, 0],
+          [5, 0],
+          [10, 0],
+          [50, 0],
+          [100, 0],
+          [500, 0],
+          [1000, 0],
+          [5000, 0],
+          [50000, 0]
+        ],
+        "sum": 0,
+        "count": 0
+      },
+      "description": "The wait time for a worker to get a connection."
+    }
```

Comment:
Missing histograms:

```suggestion
    },
    {
      "key": "query_operation_total_elapsed_time_ms",
      "labels": {},
      "value": {
        "buckets": [
          [0,0],
          [1,1],
          [5,1],
          [10,0],
          [50,0],
          [100,0],
          [500,0],
          [1000,0],
          [5000,0],
          [50000,0]
        ],
        "sum": 1.367708,
        "count": 2
      },
      "description": ""
    },
    {
      "key": "query_total_elapsed_time_ms",
      "labels": {},
      "value": {
        "buckets": [
          [0,0],
          [1,5],
          [5,0],
          [10,0],
          [50,0],
          [100,0],
          [500,0],
          [1000,0],
          [5000,0],
          [50000,0]
        ],
        "sum": 1.113832,
        "count": 5
      },
      "description": ""
    }

```

I've noticed that they also miss description, not sure if that's intentional @garrensmith